test/common/common/utility_test.cc

```diff
@@ -813,6 +814,19 @@ TEST(DateFormatter, FromTime) {
   EXPECT_EQ("aaa00", DateFormatter(std::string(3, 'a') + "%H").fromTime(time2));
 }
 
+// Check the time complexity. Make sure DateFormatter can finish parsing long messy string without
+// crashing/freezing Finish in 0-2 seconds if O(n). Finish in 30-120 seconds if O(n^2)
+TEST(DateFormatter, ParseLongString) {
```

Comment:
Seems like you want to create a benchmark instead? (see: https://github.com/envoyproxy/envoy/blob/master/test/common/common/utility_speed_test.cc). 

This case is actually good as well. However, I think here, we should set the expected formatted string too. And what do you think of using [abseil-based string-concat utils](https://abseil.io/docs/cpp/guides/strings#abslstrcat-and-abslstrappend-for-string-concatenation) instead of the one from `sstream`? Thanks!

Reply:
Thanks for the suggestion! I have replaced stringstream with absl::StrAppend().
I also added the expected result there and fixed the typos you mentioned. Thanks a lot and sorry for my mistakes.

I'm not sure whether this should be a benchmark or unit test because this test case actually caused an timeout issue in OSS-fuzz-test. And in order to benefit future regression test, I added this test case to unit test cases to avoid this issue from happening again. Do you have an idea on where to put this test case in this situation?

Reply:
With regard to use of a benchmark vs. a test:  We discussed this with @htuch and @asraa .  The consensus is that this should be a unit test, for the following reason:  If it regresses, we want a test to fail.  Marking this test as "small", and passing in a long string, ensures that any change which re-introduces the bug causes a test failure.  Benchmark regressions do not block code submission.

Reply:
That's a fair point 👍 . This probably impacts `BM_DateTimeFormatterWithSubseconds` (but with long input), as a complement of this unit test. I think, optionally providing a benchmark (hence allow to experiment by measurement) will not hurt, while the addition of it doesn't block this PR to be merged. 🙂.